File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = {'a', 'e', 'i', 'o', 'u'}
        cur = 0
        for i in range (0, k):
            if s[i] in vowels:
                cur += 1 
        max_c = cur
        for i in range (k, len(s)):
            if s[i] in vowels:
                cur += 1
            
            if s[i-k] in vowels:
                cur -= 1
            max_c = max(max_c, cur)

        return max_c

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
    
        HashSet <Character> vowels = new HashSet<Character>();
        vowels. add('a');
        vowels. add('e');
        vowels. add('i');
        vowels. add('o');
        vowels. add('u');

        char [] ch = s.toCharArray();
        int length =0;
        for (int i = 0; i < k; i++)
        {
            if(vowels.contains(ch[i]))
            {
                length += 1;
            }
        }

        int max_length = length;

        for ( int i = k; i < s.length(); i++)
        {
            if(vowels.contains(ch[i]))
            {
                length += 1;
            }

            if (vowels.contains(ch[i-k]))
            {
                length -= 1;
            }
            max_length = Math.max(length, max_length);
        }
        return max_length;
    }
}